eclass/cross.eclass

```diff
@@ -0,0 +1,83 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+# @ECLASS: cross.eclass
+# @MAINTAINER:
+# cat@catcream.org
+# @AUTHOR:
+# Alfred Persson Forsberg <cat@catcream.org> (21 Jul 2023)
+# @SUPPORTED_EAPIS: 7 8
+# @BLURB: Convenience wrappers for packages used by the Crossdev tool.
+
+inherit toolchain-funcs
+
+case ${EAPI} in
+	7|8) ;;
+	*) die "${ECLASS}: EAPI ${EAPI:-0} not supported" ;;
+esac
+
+if [[ -z ${_CROSS_ECLASS} ]]; then
+_CROSS_ECLASS=1
+
+# @ECLASS_VARIABLE: _CROSS_CATEGORY_PREFIX
+# @INTERNAL
+_CROSS_CATEGORY_PREFIX=""
+
+# @ECLASS_VARIABLE: _IS_CROSSPKG_LLVM
+# @INTERNAL
+_IS_CROSSPKG_LLVM=0
+if [[ ${CATEGORY} == cross_llvm-* ]] ; then
+	_IS_CROSSPKG_LLVM=1
+	_CROSS_CATEGORY_PREFIX="cross_llvm-"
+fi
+
+# @ECLASS_VARIABLE: _IS_CROSSPKG_GCC
+# @INTERNAL
+_IS_CROSSPKG_GCC=0
+if [[ ${CATEGORY} == cross-* ]] ; then
+	_IS_CROSSPKG_GCC=1
+	_CROSS_CATEGORY_PREFIX="cross-"
+fi
```

Comment:
Do you really need the distinction between "gcc" and "llvm" here? FWICS "gcc" case is not tested anywhere, and "llvm" is tested only in compiler-rt, and I don't see why the generic "is crossdev" check wouldn't work there.

Reply:
Indeed just a generic check would be fine in all cases currently, especially since compiler-rt isn't used by GCC crossdev. Did have some check in the musl ebuild before. Do you think I should remove the `is_crosspkg_*` functions atm?

Reply:
Yes, please. Let's keep things simple.